ext/mbstring/tests/euc_tw_encoding.phpt

```diff
@@ -25,7 +25,20 @@ findInvalidChars($fromUnicode, $invalid, $unused, array_fill_keys(range(0,0xFF),
 convertAllInvalidChars($invalid, $fromUnicode, 'UTF-16BE', 'EUC-TW', '%');
 echo "Tested UTF-16BE -> EUC-TW\n";
 
+// Test "long" illegal character markers
+mb_substitute_character("long");
+convertInvalidString("\x80", "BAD+80", "EUC-TW", "UTF-8");
+convertInvalidString("\x8E\x20", "BAD+8E20", "EUC-TW", "UTF-8");
+convertInvalidString("\x8E\xA1\x20", "BAD+8EA120", "EUC-TW", "UTF-8");
+convertInvalidString("\x8E\xA1\xA1\x20", "BAD+A1A120", "EUC-TW", "UTF-8");
```

Comment:
Is `8E` supposed to be missing here? It's not entirely obvious to me which part of the erroneous sequence we include or don't include in some cases.

Reply:
When converting bytes to wchars, we flag erroneous byte sequences by passing through a fake 'wchar' with a special bit pattern in the high bits. The low bits carry the erroneous sequence, and we generate the long subtitution from them.

Because the high bits are used to indicate that it's an error marker and not a real wchar, we can't use those high bits to carry part of the erroneous sequence. That means we can really only use the low 24 bits (a wchar is 32 bits) to hold 3 erroneous bytes. So if the erroneous sequence has 4 or more bytes, we can only show 3 of them in the long substitution.

Make sense?

Reply:
Oh yeah, that makes sense. And is possibly another argument not to bother with `BAD+` -- we can't even transfer the whole illegal sequence.

Reply:
Good point.